amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/KinesisShardDetector.java

```diff
@@ -192,9 +192,9 @@ private ListShardsResponse listShards(ShardFilter shardFilter, final String next
         exceptionManager.add(ResourceInUseException.class, t -> t);
         exceptionManager.add(KinesisException.class, t -> t);
 
-        ListShardsRequest.Builder request = KinesisRequestsBuilder.listShardsRequestBuilder().shardFilter(shardFilter);
+        ListShardsRequest.Builder request = KinesisRequestsBuilder.listShardsRequestBuilder();
         if (StringUtils.isEmpty(nextToken)) {
-            request = request.streamName(streamIdentifier.streamName());
```

Comment:
We don't need to, if a filter is null then this is equivalent to an ordinary listShards call since it uses same API but `ShardFilter` parameter is optional.